lib/src/client/dio_interceptor.dart

```diff
@@ -0,0 +1,107 @@
+import 'package:dio/dio.dart';
+import 'package:requests_signature_dart/requests_signature_dart.dart';
+
+import 'dart:convert';
+import 'package:requests_signature_dart/src/client/requests_signature_options.dart';
+
+import 'dart:typed_data';
+
+/// Interceptor for signing outgoing requests with request signature.
+///
+/// This interceptor signs the outgoing requests with a request signature
+/// before forwarding them to the inner Dio client for processing.
+class RequestsSignatureInterceptor extends Interceptor {
+  final RequestsSignatureOptions _options;
+  final ISignatureBodySourceBuilder _signatureBodySourceBuilder;
+  final ISignatureBodySigner _signatureBodySigner;
+
+  /// Constructs a new [RequestsSignatureInterceptor].
+  ///
+  /// The [options] parameter specifies the signature options.
+  ///
+  /// Optionally, you can provide custom implementations for
+  /// [signatureBodySourceBuilder] and [signatureBodySigner].
+  RequestsSignatureInterceptor(
+    this._options, {
+    ISignatureBodySourceBuilder? signatureBodySourceBuilder,
+    ISignatureBodySigner? signatureBodySigner,
+  })  : _signatureBodySourceBuilder =
+            signatureBodySourceBuilder ?? SignatureBodySourceBuilder(),
+        _signatureBodySigner = signatureBodySigner!;
```

Comment:
To replicate the .NET behavior, you need a default signer implementation in case null is passed.